yjit/src/backend/ir.rs

```diff
@@ -280,9 +282,10 @@ impl From<VALUE> for Opnd {
 #[derive(Clone, Copy, PartialEq, Eq, Debug)]
 pub enum Target
 {
-    CodePtr(CodePtr),     // Pointer to a piece of YJIT-generated code
-    SideExitPtr(CodePtr), // Pointer to a side exit code
-    Label(usize),         // A label within the generated code
+    CodePtr(CodePtr),          // Pointer to a piece of YJIT-generated code
+    SideExit(Option<Counter>), // Side exit with a counter
```

Comment:
Why wrap the counter in an `Option` here? Why not make it mandatory?

Reply:
Many exiting places are lacking a counter and adding new counters for all of them would make the diff much larger, so I'm deliberately leaving it for another PR. But once we add counters to all places, changing `SideExit(Option<Counter>)` to `SideExit(Counter)` afterwards shouldn't be difficult.

Reply:
Sounds good.